### PR TITLE
Layered navigation system

### PR DESCRIPTION
**Pull request by benborgers**
This introduces two levels of navigation: one for the entire NFToken site, and one just for the NFToken docs. 

https://share.cleanshot.com/y932U5

I’m not too sure about the design of the mobile docs nav — happy to hear suggestions for other options! 

## To do

- [ ] Scrolling down on a docs page on mobile and then navigating to another page causes the underline on the primary nav to fly up weirdly. 

## Checklist

- [ ] Splash page looks good in desktop and mobile. 
- [ ] “Learn more” and “Create an NFT” links work. 
- [ ] Sidebar and header show on docs pages and are position: sticky when scrolling. 
- [ ] Docs look good on mobile. 
- [ ] Mobile nav works even when scrolled down the page. 
- [ ] Mobile nav closes when a link is clicked. 
- [ ] Page is not scrollable while mobile nav is open. 
- [ ] Live Minting Demo works. 
- [ ] All pages look good when browser font size is increased (in settings, not cmd+/cmd-). 
- [ ] All pages look good in dark mode. 
- [ ] All pages look good in Safari. 
- [ ] All pages look good in Firefox.

### CODE REVIEW

File: docs/components/all-pages/Header.tsx [hunk 1]
```diff
@@ -1,75 +1,54 @@
...
+              className="github"
+            >
+              <span>GitHub</span>
+              <ExternalLinkIcon />
+            </a>
+          </div>
+
+          <div className="site-nav">
+            <div>
+              <Link href="/docs/overview">
```
Comment (vpontis) [inline docs/components/all-pages/Header.tsx:33]: Can we use `LuxLink` as much as possible?

File: docs/components/all-pages/Header.tsx [hunk 2]
```diff
@@ -1,75 +1,54 @@
...
+                <a>Docs</a>
+              </Link>
+              {router.pathname.startsWith("/docs") && (
+                <motion.div className="underline" layout layoutId="underline" />
+              )}
+            </div>
+            <div>
+              <Link href="/mintlists">
```
Comment (vpontis) [inline docs/components/all-pages/Header.tsx:41]: I think it makes sense to make these easier to extend

```
enum TopLevelTab {}
const TopLevelTabs = [...]

// in render
{TopLevelTabs.map()}
```

**Review by vpontis — CHANGES_REQUESTED** [review]
Where would creating a mintlist / creating an NFT live? It feels kinda weird to throw you back to the docs. Should mintlists also have a left sidepanel?

**Comment by benborgers** [comment]
Hmm we could separate out Create an NFT and Create a Mintlist into their own pages outside of the docs, and then group them under Mintlists and NFTs in the main navigation? What do you think of this kinda thing: 

2022-07-08 at 14 37 31@2x

**Comment by vpontis** [comment]
@benborgers I pushed some things here. I didn't like the Android style nav so I tried a two tiered tab system.

**Review by vpontis — APPROVED** [review]

**Comment by benborgers** [comment]
I really like it!

**Pull request merged**